django_x509/base/models.py

```diff
@@ -43,6 +44,14 @@
 }
 
 
+def datetime_to_string(datetime_):
+    """
+    Converts datetime.datetime object to UTCTime/GeneralizedTime string following RFC5280.
+    (Returns string encoded in UTCtime for dates through year 2049, otherwise in GeneralizedTime format)
+    """
+    return datetime_.strftime(utc_time) if datetime_.year < 2050 else datetime_.strftime(generalized_time)
```

Comment:
I think this before was on two lines right?

You should not opt for shortcuts when dealing with test coverage.

Please put it back to use an if with two lines of code and ensure both cases are tested, hence you need to add a new test for the new line being added. It's a bit of work but pays off greatly over time and if you want to do GSoC with us you need to learn this because we use TDD a lot to ensure our software works as expected and keeps working as expected over time.

Reply:
No, that was another function, which did conversion from string_to_datetime. It was not required since `crypto.get_notBefore().decode('utf8')` returns ASN1Time formatted like GeneralisedTime irrespective of time format in certificate. Therefore, I removed it.

I have made the required changes. :smile: :

Reply:
Coverage decreased because there were no test written for that case.

Reply:
@nemesisdesign have a look at [this](https://github.com/TheOneAboveAllTitan/django-x509/pull/2), I was not sure of how to write test for this case. Should I squash these commit or not?

Reply:
@TheOneAboveAllTitan yes send the commit into this branch from which you opened the PR so we keep all the discussion in one place. You can squash/fixup as well since it's really the same patch and more commits are not needed in this case.

From what I see in that PR, you need to assert for the presence of generalized or UTC time to be really sure the code works as expected.